polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -386,6 +388,15 @@ void MLIRScanner::init(FunctionOpInterface Func, const FunctionToEmit &FTE) {
   checkFunctionParent(Function, FTE.getContext(), Module);
 }
 
+void MLIRScanner::setEntryAndAllocBlock(Block *B) {
+  AllocationScope = EntryBlock = B;
+  Builder.setInsertionPointToStart(B);
+  // If block is linked, then the function contexts should match.
+  assert(!B->getParentOp() ||
```

Comment:
Can we put parentheses around the boolean condition part here?

Reply:
done.